### PR TITLE
Feature/vanilla-CSP-process -rules

### PR DESCRIPTION
**Pull request by ofiriro3**
This PR enables our Kubelet process rules, to support configuration file as an input.

Closes: https://github.com/elastic/security-team/issues/3279

### CODE REVIEW

File: compliance/cis_k8s/rules/cis_4_2_2/rule.rego [hunk 1]
```diff
@@ -6,20 +6,29 @@ import data.compliance.lib.common
...
-default rule_evaluation = false
-
 rule_evaluation {
+	is_authorization_allow_all
+}
+
+is_authorization_allow_all {
 	process_args["--authorization-mode"]
-	assert.is_false(common.arg_values_contains(process_args, "--authorization-mode", "AlwaysAllow"))
+	not common.contains_key_with_value(process_args, "--authorization-mode", "AlwaysAllow")
```
Comment (oren-zohar) [inline compliance/cis_k8s/rules/cis_4_2_2/rule.rego:22]: I think we should revert this change since it can contain more than one value, i.e.
```rego
rule_input("--authorization-mode=AlwaysAllow,RBAC")
```
Also, can you add tests with more than one authorization mode, please?

Comment (oren-zohar) [inline compliance/cis_k8s/rules/cis_4_2_2/rule.rego:22]: I think we should revert this change, it can have more than one authorization mode, see [here](https://kubernetes.io/docs/reference/access-authn-authz/authorization/). Also, can you add tests for this case? I think I missed it.

File: compliance/cis_k8s/rules/cis_4_2_3/rule.rego [hunk 2]
```diff
@@ -4,19 +4,41 @@ import data.compliance.cis_k8s
...
+}
+
+rule_evaluation {
+	contains_ca_file
+}
+
+# In case both flags and configuration file are specified, the executable argument takes precedence.
+# Checks that the entry for authentication:x509:clientCAFile: set to a valid path.
+rule_evaluation {
+	not contains_ca_file
```
Comment (oren-zohar) [inline compliance/cis_k8s/rules/cis_4_2_3/rule.rego:22]: I think this line is redundant

File: compliance/cis_k8s/rules/cis_4_2_3/rule.rego [hunk 3]
```diff
@@ -4,19 +4,41 @@ import data.compliance.cis_k8s
...
+	data_adapter.process_config.config.authentication.x509.clientCAFile
+	not data_adapter.process_config.config.authentication.x509.clientCAFile == ""
```
Comment (oren-zohar) [inline compliance/cis_k8s/rules/cis_4_2_3/rule.rego:24]: what is this check based on? can the path be an empty string? wouldn't the key itself be missing instead?

File: compliance/kubernetes_common/test_data.rego [hunk 4]
```diff
@@ -15,11 +15,17 @@ filesystem_input(filename, mode, uid, gid) = {
 }
 
 # genrates `process` type input data
-process_input(process_name, arguments) = {
+process_input(process_name, arguments) = result {
+	result = process_input_with_external_data(process_name, arguments, {})
+}
```
Comment (oren-zohar) [inline compliance/kubernetes_common/test_data.rego:20]:
```suggestion
process_input(process_name, arguments) = process_input_with_external_data(process_name, arguments, {})
```

File: compliance/lib/data_adapter.rego [hunk 5]
```diff
@@ -49,6 +49,11 @@ process_args = args {
 	args := {arg: value | [arg, value] = common.split_key_value(process_args_list[_])}
 }
 
+process_config = config {
+	is_process
+    config := { key: value | value = input.resource.external_data[key]}
+}
```
Comment (oren-zohar) [inline compliance/lib/data_adapter.rego:55]: 💯

**Review by oren-zohar — CHANGES_REQUESTED** [review]
Looks great! some minor comments

File: compliance/cis_k8s/rules/cis_4_2_3/rule.rego (hunk 2 above)
Comment (ofiriro3) [reply compliance/cis_k8s/rules/cis_4_2_3/rule.rego:22]: Why do you think it's redundant? 

> If the --client-ca-file argument is not present, check that there is a Kubelet config file
specified by --config, and that the file sets authentication: x509: clientCAFile to the
location of the client certificate authority file.

File: compliance/cis_k8s/rules/cis_4_2_2/rule.rego (hunk 1 above)
Comment (ofiriro3) [reply compliance/cis_k8s/rules/cis_4_2_2/rule.rego:22]: Nice almost catch - but the link you sent relevant for all the processes, kubelet behaves a little bit different [here](
https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/#:~:text=%2D%2D-,authorization,-%2Dmode%20string%C2%A0%C2%A0%C2%A0%C2%A0%C2%A0Default)

File: compliance/cis_k8s/rules/cis_4_2_3/rule.rego (hunk 2 above)
Comment (oren-zohar) [reply compliance/cis_k8s/rules/cis_4_2_3/rule.rego:22]: since if `contains_ca_file` is true `rule_evaluation` is also true, and we will not enter this check in the first place. Try removing it and see if it affects anything.

**Review by oren-zohar — APPROVED** [review]
💯
